File: fixtures/scale/swift/small/Render/Renderer.swift
import Foundation

struct Renderer {
    let options: Options
    let theme: Theme

    init(options: Options) {
        self.options = options
        self.theme = Theme.named(options.themeName)
    }

    func render(_ text: String) -> String {
        let lines = Wrapper.wrap(text, width: options.width)
        var aligned = lines.map { Alignment.alignLine($0, width: options.width, mode: options.align) }
        if options.borderStyle != "none" {
            aligned = Border(style: options.borderStyle).frame(aligned, width: options.width)
        }
        return theme.apply(aligned.joined(separator: "\n"))
    }
}

File: fixtures/scale/dart/small/lib/render/renderer.dart
import '../border/frames.dart';
import '../layout/aligner.dart';
import '../layout/wrapper.dart';
import '../model/document.dart';
import '../model/render_options.dart';
import '../style/theme.dart';
import 'border_painter.dart';

class Renderer {
  final RenderOptions options;

  Renderer(this.options);

  String render(Document document) {
    final width = options.clampedWidth;
    final body = <String>[];
    for (final line in document.lines) {
      for (final piece in wrap(line, width)) {
        body.add(align(piece, width, options.align));
      }
    }
    final frame = frameNamed(options.frameName);
    final theme = Theme.named('plain');
    final painter = BorderPainter(frame);
    return painter.paint(body, theme, width);
  }
}
